### PR TITLE
Added new method to get TZid.

### PR DESCRIPTION
**Pull request by gonzalonarbaiz**
tests are missing and we have to validate with BE team the id to be used.

## 
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-android-sdk/1270)

### CODE REVIEW

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java [hunk 1]
```diff
@@ -41,6 +41,7 @@
 import android.os.Build;
 import android.os.Bundle;
 import android.text.TextUtils;
+import android.text.format.Time;
```
Comment (jkasten2) [inline OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:44]: This import is not used, we can remove.

**Review by jkasten2 — CHANGES_REQUESTED** [review]
* Please add a unit test
* nit on extra import we don't need to include

File: OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java [hunk 2]
```diff
@@ -877,6 +877,17 @@ public void testOnSessionCalledOnlyOncePer30Sec() throws Exception {
       assertEquals("{\"app_id\":\"b4f7f966-d8cc-11e4-bed1-df8f05be55ba\"}", ShadowOneSignalRestClient.lastPost.toString());
    }
 
+   @Test
+   public void testRequestMadeWithCorrectTimeZoneID() throws Exception {
+      // Will call create
+      TimeZone.setDefault(TimeZone.getTimeZone("Europe/London"));
+      OneSignalInit();
+      threadAndTaskWait();
+      ShadowOneSignalRestClient.lastUrl = null;
```
Comment (Jeasmine) [inline OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java:886]: Why do you need this line?

File: OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java [hunk 3]
```diff
@@ -877,6 +877,17 @@ public void testOnSessionCalledOnlyOncePer30Sec() throws Exception {
...
+      JSONObject jsonObject = new JSONObject(ShadowOneSignalRestClient.lastPost.toString());
```
Comment (Jeasmine) [inline OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java:887]: no need for a new JSONObject, ShadowOneSignalRestClient.lastPost is already a JSONObject

File: OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java [hunk 4]
```diff
@@ -877,6 +877,17 @@ public void testOnSessionCalledOnlyOncePer30Sec() throws Exception {
       assertEquals("{\"app_id\":\"b4f7f966-d8cc-11e4-bed1-df8f05be55ba\"}", ShadowOneSignalRestClient.lastPost.toString());
    }
 
+   @Test
+   public void testRequestMadeWithCorrectTimeZoneID() throws Exception {
+      // Will call create
+      TimeZone.setDefault(TimeZone.getTimeZone("Europe/London"));
```
Comment (Jeasmine) [inline OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java:883]: Can you put "Europe/London" in a variable and reuse it on assertEquals?

File: OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java (hunk 3 above)
Comment (Jeasmine) [reply OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java:887]: Did you check this comment?

**Review by Jeasmine — APPROVED** [review]

**Review by gonzalonarbaiz — COMMENTED** [review]
Done

**Review by jkasten2 — APPROVED** [review]

**Pull request merged**